lib/App.server.js

```diff
@@ -152,8 +152,9 @@ App.prototype.writeScripts = function(backend, dir, options, cb) {
     if (!util.isProduction) {
       var filenames = fs.readdirSync(dir);
       for (var i = 0; i < filenames.length; i++) {
-        var item = filenames[i].split(/[-.]/);
-        if (item[0] === app.name && item[1] !== app.scriptHash) {
+        // Match and extract name, scriptHash, and extension.
+        var fileParts = filenames[i].match(/^(.+)-(\w+)(\..+)?$/);
```

Comment:
Ah, good idea. Constructing a RegExp like that won't work unless you escape `app.name`, though.

I've instead switched it to check for matching app name / hash using `indexOf`, so it doesn't need to deal with escaping or even regular expressions at all.